python/gs/ops/sddmm.py

```diff
@@ -0,0 +1,180 @@
+from itertools import product
+from .sparse import *
+import sys
+
+__all__ = ['gsddmm', 'copy_u', 'copy_v', 'copy_e']
+
+
+def reshape_lhs_rhs(lhs_data, rhs_data):
```

Comment:
Can we only support the case without broadcast for simplicity?

Reply:
This is also copied from DGL. One question is that in sampling procedure, whether the data fed into `sddmm` can be multi-dimensional? If yes, we may need to take care of this broadcast, otherwise this is not necessary.